compiler/semdata.nim

```diff
@@ -159,6 +159,7 @@ type
     exportIndirections*: HashSet[(int, int)] # (module.id, symbol.id)
     importModuleMap*: Table[int, int] # (module.id, module.id)
     lastTLineInfo*: TLineInfo
+    sideEffects*: Table[int, seq[(TLineInfo, PSym)]] # symbol.id index
```

Comment:
Looks much too expensive for any realistically sized project. What is the memory consumption like for bootstrapping?

Reply:
```./koch boot``` (take third result):
Before:
```192.331s; 839.465MiB```
After: 
```192.473s; 839.906MiB```

```time -l nim c -d=release -o=bin/nim compiler/nim``` (run twice; take second result):
Before:
```709918720  maximum resident set size```
After: 
```711106560  maximum resident set size```

Reply:
* don't measure with `./koch boot`, at least measure with `./koch boot -d:release`, but ideally should be more granular, by computing memory+time for `nim c --compileOnly compiler/nim` for a nim build before this PR vs built after this PR (both built with -d:release)
* also informative would be instrumenting code to compute the sum of `sideEffects.len` for all PContext objects created

I would guess the cost is not significant, but it should be indeed verified.

Reply:
* compile times are indistinguishable, measuring best of 4 for `nim c --compileOnly compiler/nim` before vs after PR for a nim compiled with -d:release; memory usage
* as for precise memory use here's my instrumentation:
https://gist.github.com/timotheecour/5b8b1f1602ce3876ba38b189fe14e6a6
(totNode: 13231319, totSym: 321575, sideEffectsMax: 687, sideEffectsTot: 20367)
nim --eval:'echo 13231319/20367'
649.6449648941916

=> conclusion: it's not costly.